smdebug/tensorflow/utils.py

```diff
@@ -195,35 +194,49 @@ def get_original_fetch_ops(fetches):
 """
 
 
-def is_parameter_server_strategy(tf_config: str) -> bool:
+def load_tf_config_json(tf_config: str):
     try:
-        tf_config = json.loads(tf_config)
+        return json.loads(tf_config)
     except (json.JSONDecodeError, TypeError):
-        return False  # Do not break for incorrectly set tf_config
-    return "cluster" in tf_config and "ps" in tf_config["cluster"]
+        # if tf_config is None throws TypeError, so return None from next line
+        return None
 
 
-def is_mirrored_strategy(strat):
-    return isinstance(strat, (tf.distribute.MirroredStrategy, ContribMirroredStrategy))
+def is_parameter_server_strategy(tf_config_json) -> bool:
+    try:
+        return "cluster" in tf_config_json and "ps" in tf_config_json["cluster"]
+    except TypeError:
+        # when json is None
+        return False
 
 
-def get_worker_id_from_tf_config(tf_config: str) -> str:
+def get_worker_id_from_tf_config(tf_config_json) -> str:
```

Comment:
Especially since it looks like it used to be a string before this PR and now it's a dict.

Reply:
Yes, have changed it so we only load the dict once, and if it's not a valid dict, we never call these methods